drivers/flash/flash_stm32_ospi.c

```diff
@@ -900,7 +953,7 @@ static int flash_stm32_ospi_erase(const struct device *dev, off_t addr,
 	}
 
 	/* Maximise erase size : means the complete chip */
-	if (size > dev_cfg->flash_size) {
+	if (size >= dev_cfg->flash_size) {
```

Comment:
I think this is dangerous. If sb messes up math and ends up with giving some stupid size, you are going to erase entire flash regardless of given offset.
This should be tied to `addr == 0`, and I am not sure whether `size >= dev_cfg->flash_size` is correct here and should rather be `size == dev_cfg->flash_size`, as doing anything beyond size of flash should be considered error.

Reply:
Then I prefer to let 
```
	/* Maximise erase size : means the complete chip */
	if (size > dev_cfg->flash_size) {
		size = dev_cfg->flash_size;
	}
```
And the 	address is 0 in the erasing loop `if (size == dev_cfg->flash_size) {`